⚙️ Update Node version of GitHub Workflows to v16

## Overview

* Update Node version of GitHub Workflows to v16

## Why

* When `npm run lint` in GitHub Workflows, v14 makes error.

```
Run npm run lint
  npm run lint
  shell: /usr/bin/bash -e {0}
  env:
    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
    NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX

> @openreachtech/jest-constructor-spy@1.0.1 lint /home/runner/work/jest-constructor-spy/jest-constructor-spy
> eslint .


Oops! Something went wrong! :(

ESLint: 8.41.0

/home/runner/work/jest-constructor-spy/jest-constructor-spy/node_modules/eslint-plugin-jsdoc/dist/rules/informativeDocs.js:99
    descriptionReported ||= tag.description === description && /** @type {import('comment-parser').Spec & {line: import('../iterateJsdoc.js').Integer}} */
                        ^^^

SyntaxError: Unexpected token '||='
    at wrapSafe (internal/modules/cjs/loader.js:10[2](https://github.com/openreachtech/jest-constructor-spy/pull/63/checks#step:5:2)9:16)
    at Module._compile (internal/modules/cjs/loader.js:1078:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:114[3](https://github.com/openreachtech/jest-constructor-spy/pull/63/checks#step:5:3):10)
    at Module.load (internal/modules/cjs/loader.js:979:32)
    at Function.Module._load (internal/modules/cjs/loader.js:819:12)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at Object.<anonymous> (/home/runner/work/jest-constructor-spy/jest-constructor-spy/node_modules/eslint-plugin-jsdoc/dist/index.js:21:[4](https://github.com/openreachtech/jest-constructor-spy/pull/63/checks#step:5:4)7)
    at Module._compile (internal/modules/cjs/loader.js:1114:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @openreachtech/jest-constructor-spy@1.0.1 lint: `eslint .`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @openreachtech/jest-constructor-spy@1.0.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-10-30T09_01_14_3[6](https://github.com/openreachtech/jest-constructor-spy/pull/63/checks#step:5:6)[8](https://github.com/openreachtech/jest-constructor-spy/pull/63/checks#step:5:9)Z-debug.log
Error: Process completed with exit code 2.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [14.x]
+        node-version: [16.x]
     steps:
       - uses: actions/checkout@v2
       - name: Use Node.js ${{ matrix.node-version }}

